# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/src/AsyncPanZoomController.cpp
# Commit: da6f855e06e0
# Full Hash: da6f855e06e0a75dda1a87e010447b4616cf74aa
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2021-05-08 09:56:13
# Description:
#   Bug 1710013 - Hold a strong reference of OverscrollAnimation instance to avoid being derefed. r=botond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114580
# ==============================================================================

diff -r fcaa9068465d -r da6f855e06e0 gfx/layers/apz/src/AsyncPanZoomController.cpp
--- a/gfx/layers/apz/src/AsyncPanZoomController.cpp	Fri May 07 21:30:59 2021 +0000
+++ b/gfx/layers/apz/src/AsyncPanZoomController.cpp	Fri May 07 22:32:21 2021 +0000
@@ -2681,7 +2681,7 @@
     // If there is an on-going overscroll animation, we tell the animation
     // whether the displacements should be handled by the animation or not.
     MOZ_ASSERT(mAnimation->AsOverscrollAnimation());
-    if (OverscrollAnimation* overscrollAnimation =
+    if (RefPtr<OverscrollAnimation> overscrollAnimation =
             mAnimation->AsOverscrollAnimation()) {
       overscrollAnimation->HandlePanMomentum(logicalPanDisplacement);
       // And then as a result of the above call, if the animation is currently
